src/validation/tests/variable_validation_tests.rs

```diff
@@ -158,3 +158,121 @@ fn sized_varargs_require_type() {
+                max_real : REAL :=  3.50282347E+38; //  3.40282347E+38
+
+                // f64
+                min_lreal : LREAL := -1.8076931348623157E+308; // -1.7976931348623157E+308
+                max_lreal : LREAL :=  1.8076931348623157E+308; //  1.7976931348623157E+308
+            END_VAR
+        END_FUNCTION
+        ",
+    );
+
+    assert_eq!(diagnostics.len(), 20);
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn overflow_expression() {
+    // Check if we detect overflows in expressions
+    let diagnostics = parse_and_validate(
+        "
+        FUNCTION main : DINT
+            VAR
+                // 8
+                min_sint    : SINT  := ((-128 * 1) * 2);    // -128
+                max_sint    : SINT  :=  ((127 * 1) * 2);    //  127
+
+                min_usint   : USINT  := ((1 * 1) * -2);     // 0
+                max_usint   : USINT  := ((256 * 1) * 2);    // 256
+
+                // 16
+                min_int     : INT  := ((-32_768 * 1) * 2);   // -32768
+                max_int     : INT  :=  ((32_767 * 1) * 2);   //  32767
+
+                min_uint    : UINT  := ((1 * 1) * -2);      // 0
+                max_uint    : UINT  := ((65_536 * 1) * 2);  // 65536
+
+                // 32
+                min_dint    : DINT  := ((-2_147_483_649 * 1) * 2);  // -2_147_483_648
+                max_dint    : DINT  := (( 2_147_483_648 * 1) * 2);  //  2_147_483_647
+
+                min_udint   : UDINT  := ((1 * 1) * -2);             // 0
+                max_udint   : UDINT  := ((4_294_967_296 * 1) * 2);  // 4_294_967_296
+
+                // 64
+                min_lint    : LINT  := ((-9_223_372_036_854_775_808 * 1) * 2);  // -9_223_372_036_854_775_808
+                max_lint    : LINT  := (( 9_223_372_036_854_775_807 * 1) * 2);  //  9_223_372_036_854_775_807
+
+                min_ulint   : ULINT  := ((1 * 1) * -2);                         // 0
+                max_ulint   : ULINT  := ((18_446_744_073_709_551_615 * 1) * 2); // 18_446_744_073_709_551_615
+
+                // f32
+                min_real : REAL := ((-3.40282347E+38 * 1) * 2); // -3.40282347E+38
+                max_real : REAL := (( 3.40282347E+38 * 1) * 2); //  3.40282347E+38
+
+                // f64
+                min_lreal : LREAL := ((-1.7976931348623157E+308 * 1) * 2); // -1.7976931348623157E+308
+                max_lreal : LREAL := (( 1.7976931348623157E+308 * 1) * 2); //  1.7976931348623157E+308
+            END_VAR
+        END_FUNCTION
+        ",
+    );
+
+    assert_eq!(diagnostics.len(), 20);
+    assert_validation_snapshot!(diagnostics);
+}
+
+#[test]
+fn overflow_global() {
+    // Check if we detect overflows in literals
+    let diagnostics = parse_and_validate(
+        "
+        VAR_GLOBAL
+            a : INT := 65537;
+            b : INT := 65536 + 1;
+        END_VAR
+        ",
+    );
+
+    assert_eq!(diagnostics.len(), 2);
+    assert_validation_snapshot!(diagnostics);
+}
```

Comment:
also add a test that checks for const-references
'''rust
#[test]
fn overflow_with_constants() {
    // Check if we detect overflows with constants
    let diagnostics = parse_and_validate(
        "
        VAR_GLOBAL CONSTANT
            a : INT := 6500; // ok
            b : INT := 6500; // ok
            c : INT := a + b; // not ok 
        END_VAR
        ",
    );

    assert_validation_snapshot!(diagnostics);
}
'''